rpm/containers-common.spec

```diff
@@ -0,0 +1,200 @@
+
+%description
+This package contains common configuration files and documentation for container
+tools ecosystem, such as Podman, Buildah and Skopeo.
+
+It is required because the most of configuration files and docs come from projects
+which are vendored into Podman, Buildah, Skopeo, etc. but they are not packaged
+separately.
+
+%package extra
+Summary: Extra dependencies for Podman and Buildah
+Requires: %{name} = %{epoch}:%{version}-%{release}
+Requires: container-network-stack
+Requires: oci-runtime
+Conflicts: podman < 5:5.0.0~rc4-1
+Recommends: composefs
+Recommends: crun
+Requires: (crun if fedora-release-identity-server)
+Requires: netavark >= 1.10.3-1
+Suggests: slirp4netns
+Requires: passt
+Requires: iptables
+Requires: nftables
+Recommends: qemu-user-static
+Requires: (qemu-user-static-aarch64 if fedora-release-identity-server)
+Requires: (qemu-user-static-arm if fedora-release-identity-server)
+Requires: (qemu-user-static-x86 if fedora-release-identity-server)
+
+%description extra
+This subpackage will handle dependencies common to Podman and Buildah which are
+not required by Skopeo.
+
+%prep
+%autosetup -Sgit %{name}-%{version_no_tilde}
+
+# Fine-grain distro- and release-specific tuning of config files,
+# e.g., seccomp, composefs, registries on different RHEL/Fedora versions
+bash rpm/update-config-files.sh
+
+%build
+
+%install
+# install config and policy files for registries
+install -dp %{buildroot}%{_sysconfdir}/containers/{certs.d,oci/hooks.d,systemd}
+install -dp %{buildroot}%{_sharedstatedir}/containers/sigstore
+install -dp %{buildroot}%{_datadir}/containers/systemd
+install -dp %{buildroot}%{_prefix}/lib/containers/storage
+install -dp -m 700 %{buildroot}%{_prefix}/lib/containers/storage/overlay-images
+touch %{buildroot}%{_prefix}/lib/containers/storage/overlay-images/images.lock
+install -dp -m 700 %{buildroot}%{_prefix}/lib/containers/storage/overlay-layers
+touch %{buildroot}%{_prefix}/lib/containers/storage/overlay-layers/layers.lock
+
+install -Dp -m0644 %{SOURCE9} %{buildroot}%{_sysconfdir}/containers/registries.conf
+install -Dp -m0644 %{SOURCE10} %{buildroot}%{_sysconfdir}/containers/registries.conf.d/000-shortnames.conf
+install -Dp -m0644 %{SOURCE11} %{buildroot}%{_sysconfdir}/containers/registries.d/default.yaml
+install -Dp -m0644 %{SOURCE12} %{buildroot}%{_sysconfdir}/containers/policy.json
+install -Dp -m0644 %{SOURCE14} %{buildroot}%{_datadir}/containers/storage.conf
+
+# RPM-GPG-KEY-redhat-release already exists on rhel envs, install only on
+# fedora and centos
+%if %{defined fedora} || %{defined centos}
+install -Dp -m0644 %{SOURCE15} %{buildroot}%{_sysconfdir}/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+%endif
+
+install -Dp -m0644 contrib/redhat/registry.access.redhat.com.yaml -t %{buildroot}%{_sysconfdir}/containers/registries.d
+install -Dp -m0644 contrib/redhat/registry.redhat.io.yaml -t %{buildroot}%{_sysconfdir}/containers/registries.d
+
+# install manpages
```

Comment:
I like the removal of all the `cp` commands, but the replacement is really hairy: super-duper-duplicationy (duplicating filename and the full path). It also skips using `install -m0644`, hence relies on umask. What would you think of
```console
mkdir -p man5
for i in *.5.md; do
    go-md2man -in $i -out man5/$(basename $i .md)
done

....

for i in man5/*.5; do
    install -Dp -m0644 $i -t %{buildroot}%{_mandir}/man5
done
ln -s containerignore.5 %{buildroot}%{_mandir}/man5/.containerignore.5
```

Reply:
> ```
> mkdir -p man5
> for i in *.5.md; do
>     go-md2man -in $i -out man5/$(basename $i .md)
> done
> ```

This will need the `cp` statements added back for all those manpages from other repos listed as SourceN files, otherwise it's going to complain those files aren't found.

One thing I could do would be to move all the `go-md2man` steps to `%build` stage and then run the install loop in `%install`.
>```
> for i in man5/*.5; do
>    install -Dp -m0644 $i -t %{buildroot}%{_mandir}/man5
> done
> ln -s containerignore.5 %{buildroot}%{_mandir}/man5/.containerignore.5
>```

Reply:
> > ```
> > mkdir -p man5
> > for i in *.5.md; do
> >     go-md2man -in $i -out man5/$(basename $i .md)
> > done
> > ```
> 
> This will need the `cp` statements added back for all those manpages from other repos listed as SourceN files, otherwise it's going to complain those files aren't found.
> 

Trying something in a new commit. Let's see how that goes.

Reply:
@edsantiago You're gonna end up with some or the other `cp` or `mv` somewhere to operate on the SourceN files. So, let me know how you'd like to proceed. I could cp / mv the SourceN to docs, build all docs in `%build`, and then install them in `%install` if that seems ok.

Reply:
I like this version much, much better: it's less duplicationy. With your move of `constainers-storage.conf.5.md` to be contiguous to all the other `.md` files, this is less fragile. LGTM.